samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -172,6 +174,17 @@ def __init__(
         self._global_parameter_overrides: Optional[Dict] = None
         if aws_region:
             self._global_parameter_overrides = {"AWS::Region": aws_region}
+        
+        # Attempt to get the Account ID from the current session (if any)
+        sts = boto3.client("sts")
+        try:
+            account_id = sts.get_caller_identity()["Account"]            
+            if account_id:
+                if self._global_parameter_overrides is None:
+                    self._global_parameter_overrides = {}
+                self._global_parameter_overrides["AWS::AccountId"] = account_id
+        except TokenRetrievalError:
+            LOG.warning( "No current session found, using default AWS::AccountId" )
```

Comment:
nit: Can we move this logic into its own function? It should also make it easier to test.

Reply:
I've moved the code to its own function. Happy to write unit tests but I would appreciate some pointers on where to start, as the functionality relies on whether the user running the code is/isn't logged into an AWS account. I'm not sure how to mock the response from `sts.get_caller_identity`

Reply:
You'll want to write the tests in this file https://github.com/aws/aws-sam-cli/blob/develop/tests/unit/commands/local/cli_common/test_invoke_context.py. There should be a few examples in our unit tests that demonstrate mocking external calls. Here's on you can use as reference: https://github.com/aws/aws-sam-cli/blob/develop/tests/unit/commands/local/cli_common/test_invoke_context.py#L1355

Reply:
I've made the suggested changes (as you probably noticed Python is not my first language!) and added unit tests to make sure the new function works both with and without a called id being returned by STS